### ISSUE TITLE
bug: lib/commands/command-exec.bash improperly attempts to reference `shim_args[@]`

### ISSUE THREAD

**Issue by dossy** [asdf-vm/asdf#1810]
### Describe the Bug

Fix unbound variable error when executing shims with no arguments. When executing certain shims (like `gke-gcloud-auth-plugin`) with no arguments, the following error occurs:

```bash
/usr/local/opt/asdf/libexec/lib/commands/command-exec.bash: line 23: shim_args[@]: unbound variable
```

The issue occurs because the script attempts to reference `shim_args[@]` without properly checking if the array has been initialized or is empty.

Previously:

* https://github.com/asdf-vm/asdf/issues/522
* https://github.com/jthegedus/asdf-gcloud/issues/68
* https://github.com/Jakski/asdf-janet/pull/3
* https://github.com/code-lever/asdf-rust/issues/17

### Steps to Reproduce

1. Install asdf v0.14.1
2. Install gcloud plugin: `asdf plugin add gcloud`
3. Install gcloud: `asdf install gcloud 503.0.0`
4. Install kubectl plugin: `asdf plugin add kubectl`
5. Install kubectl: `asdf install kubectl 1.32.0`
6. Set up a GKE cluster
7. Try to access cluster: `kubectl get pods`
8. Observe error messages about unbound variable

### Expected Behaviour

`kubectl` outputs information about the pods in the cluster

### Actual Behaviour

The following error is displayed:

```
/usr/local/opt/asdf/libexec/lib/commands/command-exec.bash: line 23: shim_args[@]: unbound variable
E1216 19:50:38.362751   24048 memcache.go:265] "Unhandled Error" err="couldn't get current server API group list: Get \"https://[redacted]/api?timeout=32s\": getting credentials: exec: executable gke-gcloud-auth-plugin failed with exit code 1"
/usr/local/opt/asdf/libexec/lib/commands/command-exec.bash: line 23: shim_args[@]: unbound variable
E1216 19:50:38.480657   24048 memcache.go:265] "Unhandled Error" err="couldn't get current server API group list: Get \"https://[redacted]/api?timeout=32s\": getting credentials: exec: executable gke-gcloud-auth-plugin failed with exit code 1"
/usr/local/opt/asdf/libexec/lib/commands/command-exec.bash: line 23: shim_args[@]: unbound variable
E1216 19:50:38.596852   24048 memcache.go:265] "Unhandled Error" err="couldn't get current server API group list: Get \"https://[redacted]/api?timeout=32s\": getting credentials: exec: executable gke-gcloud-auth-plugin failed with exit code 1"
/usr/local/opt/asdf/libexec/lib/commands/command-exec.bash: line 23: shim_args[@]: unbound variable
E1216 19:50:38.715038   24048 memcache.go:265] "Unhandled Error" err="couldn't get current server API group list: Get \"https://[redacted]/api?timeout=32s\": getting credentials: exec: executable gke-gcloud-auth-plugin failed with exit code 1"
/usr/local/opt/asdf/libexec/lib/commands/command-exec.bash: line 23: shim_args[@]: unbound variable
E1216 19:50:38.825955   24048 memcache.go:265] "Unhandled Error" err="couldn't get current server API group list: Get \"https://[redacted]/api?timeout=32s\": getting credentials: exec: executable gke-gcloud-auth-plugin failed with exit code 1"
Unable to connect to the server: getting credentials: exec: executable gke-gcloud-auth-plugin failed with exit code 1
```

### Environment

```shell
$ asdf info
OS:
Darwin focus.panoptic.com 22.6.0 Darwin Kernel Version 22.6.0: Mon Jun 24 01:25:37 PDT 2024; root:xnu-8796.141.3.706.2~1/RELEASE_X86_64 x86_64

SHELL:
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin22)
Copyright (C) 2007 Free Software Foundation, Inc.

BASH VERSION:
3.2.57(1)-release

ASDF VERSION:
v0.14.1

ASDF INTERNAL VARIABLES:
ASDF_DEFAULT_TOOL_VERSIONS_FILENAME=.tool-versions
ASDF_DATA_DIR=/Users/dossy/.asdf
ASDF_DIR=/usr/local/opt/asdf/libexec
ASDF_CONFIG_FILE=/Users/dossy/.asdfrc

ASDF INSTALLED PLUGINS:
gcloud                       https://github.com/jthegedus/asdf-gcloud.git main 14d1c0d
golang                       https://github.com/kennyp/asdf-golang.git master 21d9c32
helm                         https://github.com/Antiarchitect/asdf-helm.git master a39e17b
helmfile                     https://github.com/feniix/asdf-helmfile master 3f1681e
kubectl                      https://github.com/Banno/asdf-kubectl master cbe6df4
nodejs                       https://github.com/asdf-vm/asdf-nodejs.git master 644ada3
pnpm                         https://github.com/jonathanmorley/asdf-pnpm.git master ea033db
python                       https://github.com/danhper/asdf-python.git master 8505457
ruby                         https://github.com/asdf-vm/asdf-ruby.git master 4e5b177
sops                         https://github.com/feniix/asdf-sops master 954411c
sqlite                       https://github.com/cLupus/asdf-sqlite.git main 409f677
terraform                    https://github.com/Banno/asdf-hashicorp.git master c44c314
terragrunt                   https://github.com/ohmer/asdf-terragrunt.git main 4a6651a
yarn                         https://github.com/twuni/asdf-yarn.git main 376c540
```

### asdf plugins affected (if relevant)

gcloud, kubectl

**Comment by Stratus3D** [comment]
Hi @dossy , thanks for opening the issue. I'm sorry you are seeing errors when using asdf. It's not clear to me why you are seeing this. Does this issue occur with any other commands or asdf plugins? How did you install kubectl?

**Comment by dossy** [comment]
> Hi @dossy , thanks for opening the issue. I'm sorry you are seeing errors when using asdf. It's not clear to me why you are seeing this. Does this issue occur with any other commands or asdf plugins? How did you install kubectl?

I installed kubectl with `asdf plugin add kubectl; asdf install kubectl` - I'll update the STR above to clarify.

**Comment by Stratus3D** [comment]
Hi @dossy I recently merged #1805 is a rewrite of asdf in Golang. Thanks for the PR for this issue, but I've been testing the Go code and have not encountered an issue similar to this one. Given that Go has proper slice/array types I don't expect this bug to be present in the Go version. If you want to test the Go code you can download the latest dev binaries here - https://github.com/asdf-vm/asdf/releases/tag/54d15e3.
